tools/pythonpkg/tests/sqllogic/test_sqllogic.py

```diff
@@ -0,0 +1,131 @@
+            "aws",
+            "autocomplete",
+            "excel",
+            "fts",
+            "httpfs",
+            "json",
+            "parquet",
+            "postgres_scanner",
+            "sqlsmith",
+            "sqlite_scanner",
+            "tpcds",
+            "tpch",
+            # "azure",
+            # "inet",
+            # "icu",
+            # "spatial",
+            # TODO: table function isnt always autoloaded so test fails
+        ]
+
+    def get_test_directory(self) -> str:
+        test_directory = TEST_DIRECTORY_PATH
+        if not os.path.exists(test_directory):
+            os.makedirs(test_directory)
+        return test_directory
+
+    def delete_database(self, path):
+        def test_delete_file(path):
+            try:
+                if os.path.exists(path):
+                    os.remove(path)
+            except FileNotFoundError:
+                pass
+
+        # FIXME: support custom test directory
+        test_delete_file(path)
+        test_delete_file(path + ".wal")
+
+    def execute_test(self, test: SQLLogicTest) -> ExecuteResult:
+        self.reset()
+        self.test = test
+        self.original_sqlite_test = self.test.is_sqlite_test()
+
+        # Top level keywords
+        keywords = {'__TEST_DIR__': self.get_test_directory(), '__WORKING_DIRECTORY__': os.getcwd()}
+
+        def update_value(_: SQLLogicContext) -> Generator[Any, Any, Any]:
+            # Yield once to represent one iteration, do not touch the keywords
+            yield None
+
+        self.database = SQLLogicDatabase(':memory:', None)
+        pool = self.database.connect()
+        context = SQLLogicContext(pool, self, test.statements, keywords, update_value)
+        pool.initialize_connection(context, pool.get_connection())
+        # The outer context is not a loop!
+        context.is_loop = False
+
+        try:
+            context.verify_statements()
+            res = context.execute()
+        except TestException as e:
+            res = e.handle_result()
+            if res.type == ExecuteResult.Type.SKIPPED:
+                pytest.skip(str(e.message))
+            else:
+                pytest.fail(str(e.message), pytrace=False)
+
+        self.database.reset()
+
+        # Clean up any databases that we created
+        for loaded_path in self.loaded_databases:
+            if not loaded_path:
+                continue
+            # Only delete database files that were created during the tests
+            if not loaded_path.startswith(self.get_test_directory()):
+                continue
+            os.remove(loaded_path)
+        return res
+
+
+def test_sqllogic(test_script_path: pathlib.Path):
+    gc.collect()
+    sql_parser = SQLLogicParser()
+    try:
+        test = sql_parser.parse(str(test_script_path))
+    except SQLParserException as e:
+        pytest.skip("Failed to parse SQLLogic script: " + str(e.message))
+        return
+
+    executor = SQLLogicTestExecutor()
```

Comment:
This is missing the clearing of the temp directory
```suggestion
    executor = SQLLogicTestExecutor() # FIXME: where is the `args.build_dir` equivalent?
    if os.path.exists(TEST_DIRECTORY_PATH):
        shutil.rmtree(TEST_DIRECTORY_PATH)
```

Reply:
Made a comment about build directory, this is needed for this (from `result.py`):
```py
    def get_extension_path(self, extension: str):
        if self.runner.build_directory is None:
            self.skiptest("Tried to load an extension, but --build-dir was not set!")
        root = self.runner.build_directory
        path = os.path.join(root, "extension", extension, f"{extension}.duckdb_extension")
        return path
```